File: app/src/main/java/ru/cities/game/fragment/ResultFragment.java
package ru.cities.game.fragment;

import android.content.Context;
import android.content.SharedPreferences;
import android.media.MediaPlayer;
import android.os.Bundle;
import android.view.View;
import android.widget.Button;
import android.widget.ImageView;
import android.widget.TextView;

import androidx.annotation.NonNull;
import androidx.annotation.Nullable;
import androidx.fragment.app.Fragment;

import ru.cities.game.R;

public class ResultFragment extends Fragment {
    private Button btnReplay;
    private TextView tvScore, tvRecord, tvTitle;
    private ImageView ivWinnerIcon;
    private int resId = R.raw.defeat;

    public ResultFragment() {
        super(R.layout.result_fragment);
    }

    private void initViews() {
        View view = getView();
        assert view != null;
        tvScore = view.findViewById(R.id.tv_score);
        tvRecord = view.findViewById(R.id.tv_record);
        tvTitle = view.findViewById(R.id.tv_title);
        ivWinnerIcon = view.findViewById(R.id.iv_winner_icon);
        btnReplay = view.findViewById(R.id.btn_replay);
    }
    @Override
    public void onViewCreated(@NonNull View view, @Nullable Bundle savedInstanceState) {
        super.onViewCreated(view, savedInstanceState);
        initViews();
        SharedPreferences prefs = requireActivity().getSharedPreferences("preferences", Context.MODE_PRIVATE);
        tvScore.setText(getString(R.string.score, prefs.getInt("score", 0)));
        tvRecord.setText(getString(R.string.record, prefs.getInt("record", 0)));
        btnReplay.setOnClickListener(v -> requireActivity().getSupportFragmentManager().beginTransaction()
                .replace(R.id.container, MainFragment.class, null)
                .commit());
        if (prefs.getBoolean("isVictory", false)) {
            resId = R.raw.victory;
            tvTitle.setText(getString(R.string.you_won));
            ivWinnerIcon.setImageResource(R.drawable.ic_user_won);
        }
    }

    @Override
    public void onStart() {
        super.onStart();
        MediaPlayer.create(requireActivity(), resId).start();
    }
}

File: app/src/main/java/ru/cities/game/fragment/ResultFragment.kt
package ru.cities.game.fragment

import android.content.Context
import android.os.Bundle
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import androidx.fragment.app.Fragment
import ru.cities.game.R
import ru.cities.game.databinding.ResultFragmentBinding
import ru.cities.game.util.*

class ResultFragment : Fragment(R.layout.result_fragment) {

    private var _binding: ResultFragmentBinding? = null
    private val binding get() = _binding!!

    override fun onCreateView(
        inflater: LayoutInflater,
        container: ViewGroup?,
        savedInstanceState: Bundle?
    ): View {
        setStatusBar(requireActivity(), 8, R.color.white, true)
        _binding = ResultFragmentBinding.inflate(inflater, container, false)
        return binding.root
    }

    override fun onDestroyView() {
        super.onDestroyView()
        _binding = null
    }

    override fun onViewCreated(view: View, savedInstanceState: Bundle?) {
        super.onViewCreated(view, savedInstanceState)
        val args = requireArguments()
        val score = args.get(SCORE_ID)
        val isVictory = args.get(IS_VICTORY_ID) as Boolean
        binding.tvScore.text = getString(R.string.score, score)
        val prefs = requireActivity().getSharedPreferences(PREFERENCES_FILE, Context.MODE_PRIVATE)
        binding.tvRecord.text = getString(R.string.record, prefs.getInt(KEY_SCORE, 0))
        if (isVictory) {
            binding.tvTitle.text = getString(R.string.you_won)
            binding.ivWinnerIcon.setImageResource(R.drawable.ic_user_won)
        }
        binding.btnReplay.setOnClickListener {
            requireActivity().supportFragmentManager.beginTransaction()
                .replace(R.id.container, MainFragment()).commit()
        }
    }
}
